engine/rpcserver.go

```diff
@@ -696,14 +696,14 @@ func (s *RPCServer) GetAccountInfoStream(r *gctrpc.GetAccountInfoRequest, stream
 	}()
 
 	for {
-		data, ok := <-pipe.Channel()
+		_, ok := <-pipe.Channel()
 		if !ok {
 			return errDispatchSystem
 		}
 
-		holdings, ok := data.(*account.Holdings)
-		if !ok {
-			return common.GetTypeAssertError("*account.Holdings", data)
+		holdings, err := exch.GetCachedAccountInfo(stream.Context(), assetType)
+		if err != nil {
+			return err
 		}
```

Comment:
Suggestion: If we move this call and the accounts info send above the pipe we can remove the initial code outside of this for loop for fetching the initial holdings.

Reply:
I’ll use a `init` flag inside the loop to handle the first fetch, removing the initial fetch outside the loop.